Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -238,12 +235,15 @@ public final class KafkaProducer: Service, Sendable {
         self.stateMachine.withLockedValue { $0.finish() }
     }
 
-    /// Send messages to the Kafka cluster asynchronously. This method is non-blocking.
-    /// Message send results shall be handled through the ``KafkaMessageAcknowledgements`` `AsyncSequence`.
+    /// Send a ``KafkaProducerMessage`` to the Kafka cluster.
+    ///
+    /// This method does not wait until the message is sent and acknowledged by the cluster.
+    /// Instead, it buffers the message and returns immediately.
+    /// The message will be sent out with the next batch of messages.
     ///
-    /// - Parameter message: The ``KafkaProducerMessage`` that is sent to the KafkaCluster.
-    /// - Returns: Unique ``KafkaProducerMessageID``matching the ``KafkaAcknowledgedMessage/id`` property
-    /// of the corresponding ``KafkaAcknowledgedMessage``.
+    /// - Parameter message: The ``KafkaProducerMessage`` to send.
+    /// - Returns: Unique ``KafkaProducerMessageID``matching the ``KafkaDeliveryReport/id`` property
+    /// of the corresponding ``KafkaDeliveryReport``.
     /// - Throws: A ``KafkaError`` if sending the message failed.
     @discardableResult
     public func send<Key, Value>(_ message: KafkaProducerMessage<Key, Value>) throws -> KafkaProducerMessageID {
```

Comment:
I was thinking that we have two places to receive error: 
1. Here `send` method can throw an exception
2. And `.failure` can be received in `events`

Would be nice to unite these things

Reply:
I don't think we can or should unite those two. The failure on `send` here is when we fail to enqueue a message. The failure on the events is if we failed to deliver/acknowledge the message.